torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -0,0 +1,67 @@
+"""
+Utils for caching the outputs of AOTAutograd
+"""
+from __future__ import annotations
+
+import logging
+
+import torch
+from torch._inductor.codecache import _ident, FxGraphCachePickler
+
+from .schemas import AOTConfig  # noqa: F401
+
+log = logging.getLogger(__name__)
+
+
+class AOTAutogradCacheDetails:
+    """
+    Object to capture all the details for a dynamo graph module relevant to computing
+    a safe and stable cache key for AOTAutograd.
+    """
+
+    def __init__(self, gm: torch.fx.GraphModule, config: AOTConfig):
+        self.gm = gm  # TODO: we'll handle different parts of the graph module
+        # TODO: We'll want to handle the full_args passed in as well
+        self.config = config  # Gets reduced by the Pickler
+
+    def debug_str(self) -> str:
+        return AOTAutogradCachePickler.debug_str(self)
+
+
+def _reduce_aot_config(config: AOTConfig):
+    """
+    Reduce the config to a stable key for caching.
+    """
+    return (
+        _ident,
+        (
+            config.num_params_buffers,
+            config.keep_inference_input_mutations,
+            config.is_export,
+            config.no_tangents,
+            config.dynamic_shapes,
+            config.aot_autograd_arg_pos_to_source,
+            config.enable_log,
+            config.pre_dispatch,
+        ),
+    )
+
+
+class AOTAutogradCachePickler(FxGraphCachePickler):
+    dispatch_table = FxGraphCachePickler.dispatch_table.copy()
+    dispatch_table[AOTConfig] = _reduce_aot_config
+
+
+def autograd_cache_hash(
+    gm: torch.fx.GraphModule,
+    config: AOTConfig,
```

Comment:
Hmm, I think caching on the AOTConfig object is mostly reasonable. But just calling out: AOTConfig isn't an input to AOTAutograd - it's mostly a convenience object to bundle up and pass around state about what AOTAutograd should do during tracing, and it contains a mix of inputs to AOTAutograd, and other stuff that is inferred implicitly (`is_export=False`, `aot_id=...`).

Just looking at some of the stuff in it:

`aot_id`: Including `AOTConfig.aot_id` will probably give us some false negatives (this is just a counter used to uniquely identify each compiled region in a python process).

`is_export`: AOTAutograd is also used in an export setting where we don't need the warm cache, so this will just always be set to false. `num_params_buffers`/`dynamic_shapes`: these are both values that are inferred from looking at the dynamo graph. Then again these are all cheap to cache and won't cause false negatives, so caching on them probably won't matter.